package/MDAnalysis/core/selection.py

```diff
@@ -600,6 +600,38 @@ def apply(self, group):
         return group[group.aromaticities].unique
 
 
+class SmartsSelection(Selection):
+    """Select atoms based on SMARTS queries"""
+    token = 'smarts'
+
+    def __init__(self, parser, tokens):
+        pattern = []
+        while not is_keyword(tokens[0]) or tokens[0] in ["(", ")"]:
+            val = tokens.popleft()
+            pattern.append(val)
+        self.pattern = "".join(pattern)
+
+    def apply(self, group):
+        try:
+            from rdkit import Chem
+        except ImportError:
+            raise ImportError("RDKit is required for SMARTS-based atom "
+                              "selection but it's not installed. Try "
+                              "installing it with \n"
+                              "conda install -c conda-forge rdkit")
+        pattern = Chem.MolFromSmarts(self.pattern)
+        if not pattern:
+            raise ValueError("{!r} is not a valid SMARTS query".format(
+                self.pattern))
+        mol = group.convert_to("RDKIT")
+        matches = mol.GetSubstructMatches(pattern, useChirality=True)
+        indices = [idx for match in matches for idx in match]
+        indices = [mol.GetAtomWithIdx(i).GetIntProp("_MDAnalysis_index")
```

Comment:
Can we not predict what idx RDKit will assign? Ie do the atoms sometimes get reordered?

Reply:
They get reordered when using reactions to "standardize" some functional groups like nitro, sulfone...etc into the correct form. I could also force reordering atoms at the end of the conversion but that requires building a new molecule again I think.